esp/services/ws_esdlconfig/ws_esdlconfigservice.cpp

```diff
@@ -1310,13 +1357,25 @@ bool CWsESDLConfigEx::onDeleteESDLBinding(IEspContext &context, IEspDeleteESDLBi
 
 bool CWsESDLConfigEx::onGetESDLDefinition(IEspContext &context, IEspGetESDLDefinitionRequest&req, IEspGetESDLDefinitionResponse &resp)
 {
+    if (!context.validateFeatureAccess(FEATURE_URL, SecAccess_Read, false))
+        throw MakeStringException(ECLWATCH_ROXIE_QUERY_ACCESS_DENIED, "Failed to fetch ESDL definition. Permission denied.");
+
     StringBuffer id = req.getId();
     StringBuffer definition;
     resp.setId(id.str());
+    VStringBuffer message("Successfully fetched ESDL Defintion: %s from Dali.", id.str());
+    int respcode = 0;
+
+    double ver = context.getClientVersion();
 
     try
     {
         fetchESDLDefinitionFromDaliById(id.toLowerCase(), definition);
+        if (definition.length() == 0 )
+        {
+            respcode = -1;
+            message.append("\nDefinition appears to be empty!");
```

Comment:
Yes... because we were able to fetch the definition, it just turned out to be empty...